// gemini-configparser/lib/errors.js
'use strict';

var _get = require('babel-runtime/helpers/get')['default'];

var _inherits = require('babel-runtime/helpers/inherits')['default'];

var _classCallCheck = require('babel-runtime/helpers/class-call-check')['default'];

Object.defineProperty(exports, '__esModule', {
    value: true
});

var MissingOptionError = (function (_Error) {
    _inherits(MissingOptionError, _Error);

    function MissingOptionError(optionName) {
        _classCallCheck(this, MissingOptionError);

        var message = optionName + ' is required';
        _get(Object.getPrototypeOf(MissingOptionError.prototype), 'constructor', this).call(this, message);
        this.name = 'MissingOptionError';
        this.message = message;
        this.optionName = optionName;

        Error.captureStackTrace(this, MissingOptionError);
    }

    return MissingOptionError;
})(Error);

exports.MissingOptionError = MissingOptionError;

var UnknownKeysError = (function (_Error2) {
    _inherits(UnknownKeysError, _Error2);

    function UnknownKeysError(keys) {
        _classCallCheck(this, UnknownKeysError);

        var message = 'Unknown options: ' + keys.join(', ');
        _get(Object.getPrototypeOf(UnknownKeysError.prototype), 'constructor', this).call(this, message);
        this.name = 'UnknownKeysError';
        this.message = message;
        this.keys = keys;

        Error.captureStackTrace(this, UnknownKeysError);
    }

    return UnknownKeysError;
})(Error);

exports.UnknownKeysError = UnknownKeysError;
